doc/Iron.rst

```diff
@@ -0,0 +1,92 @@
+Iron to Jazzy
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Adaption of Command-/StateInterfaces
+***************************************
+Changes from `(PR #1240) <https://github.com/ros-controls/ros2_control/pull/1240>`__
```

Comment:
That PR isn't going to be merged but was split instead. Should we reference this current PR here?

Reply:
I think there is a lot of discussion and it summarizes the changes and gives good overview. So i would vote for a reference of it